eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -127,28 +119,56 @@ private Collection<String> discover(final Path file) {
      * Get a unique list of object names from given XML.
      *
      * @param xml XML.
+     * @param tojo Current tojo.
      * @return Object names.
      */
-    private static Set<String> names(final XML xml) {
+    private Collection<String> names(final XML xml, final ForeignTojo tojo) {
         return new SetOf<>(
-            new Filtered<>(
-                obj -> !obj.isEmpty(),
-                xml.xpath(
-                    String.join(
-                        "",
-                        "//o[",
-                        "not(starts-with(@base,'.'))",
-                        " and @base != 'Q'",
-                        " and @base != '^'",
-                        " and @base != '$'",
-                        " and @base != '&'",
-                        " and not(@ref)",
-                        "]/string-join((@base, @ver),'",
-                        OnReplaced.DELIMITER,
-                        "')"
+            new Mapped<>(
+                (String name) -> this.versioned(name, tojo).toString(),
+                new Filtered<>(
+                    name -> !name.isEmpty(),
+                    xml.xpath(
+                        String.join(
+                            "",
+                            "//o[",
+                            "not(starts-with(@base,'.'))",
+                            " and @base != 'Q'",
+                            " and @base != '^'",
+                            " and @base != '$'",
+                            " and @base != '&'",
+                            " and not(@ref)",
+                            "]/string-join((@base,@ver),'",
+                            OnReplaced.DELIMITER,
+                            "')"
+                        )
                     )
                 )
             )
         );
     }
+
+    /**
+     * Handle versioning of given object name.
+     *
+     * @param name Object name with tag on not.
+     * @param tojo Current tojo.
+     * @return Versioned object name.
+     */
+    private ObjectName versioned(final String name, final ForeignTojo tojo) {
```

Comment:
@maxonfjvipon You pass `tojo` into three methods in order to find "default" version of the `tojo`  in the last method. Maybe we can do it earlier and pass only "default version" instead of "tojo" - in this case code might become clearly and more understandable. What do you think?